metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -1343,17 +1297,49 @@ def visit_join_to_time_spine_node(self, node: JoinToTimeSpineNode[SourceDataSetT
             metric_instances=parent_data_set.instance_set.metric_instances,
             metadata_instances=parent_data_set.instance_set.metadata_instances,
         )
-        table_alias_to_instance_set = OrderedDict(
-            {time_spine_alias: time_spine_dataset.instance_set, parent_alias: non_metric_time_parent_instance_set}
+
+        assert (
+            len(time_spine_dataset.instance_set.time_dimension_instances) == 1
+            and len(time_spine_dataset.sql_select_node.select_columns) == 1
+        ), "Time spine dataset not configured properly. Expected exactly one column."
+        original_time_dim_instance = time_spine_dataset.instance_set.time_dimension_instances[0]
+        time_spine_select_expr: Union[
+            SqlColumnReferenceExpression, SqlDateTruncExpression
+        ] = SqlColumnReferenceExpression(
+            SqlColumnReference(table_alias=time_spine_alias, column_name=original_time_dim_instance.spec.qualified_name)
         )
 
+        # Add requested granularity to time spine spec & column.
+        # Time spine defaults to DAY, so DAY granularity does not require any change here.
+        if node.metric_time_dimension_spec.time_granularity == TimeGranularity.DAY:
```

Comment:
Let's not hard-code DAY as the default. The time_spine_source should have the granularity, so we can update this logic to match what we had earlier, assuming we don't want to just revert back.